scheduler/test/cook/test/scheduler/constraints.clj

```diff
@@ -347,3 +347,14 @@
           (is (= [true nil] (constraints/job-constraint-evaluate constraint nil {"HOSTNAME" "hostC"})))
           ;; unsuitable
           (is (= [false "Host is not suitable for datasets"] (constraints/job-constraint-evaluate constraint nil {"HOSTNAME" "hostB"}))))))))
+
+(deftest test-job->previous-hosts-to-avoid
```

Comment:
Nitpick: The assertion also needs to validate on length to confirm duplicates are not returned.

Reply:
:man_facepalming: Good call, adding this in PR #1479.